### ISSUE TITLE
Missing data in over_time output file

### ISSUE THREAD

**Issue by ErikPoppleton** [vgapsys/g_lomepro#3]
I am trying to get the thickness of a simulation of a DOPC membrane over time.  I first made an index file with all the phosphates and lipid tails selected (groups 'PA' and 'OL') and then used that as input for `g_lomepro`:

```{ echo -e "PA"; echo -e "OL"; } | ~/software/g_lomepro/g_lomepro_static -b 0 -e 500000 -f md.xtc -n lomepro.ndx -s md.gro -apl apl1 -thick thick1 -curve curve1 -lip_num 898 -nonflat -binx 100 -biny 100```

However in the `thick1.out_over_time.dat` file, there are not always 898 entries per time step.  Can you explain why in some time steps some of the phosphate IDs are missing from the thickness?

Also am I correct that when I choose the `-nonflat` option, I'm supposed to have a group containing the whole lipid tails, or is it supposed to be just the last atom in each tail?  The paper and the help say different things.

**Comment by vgapsys** [comment]
Hey Erik,
so far I was not able to reproduce this. I added a folder "examples/thickness" with a small example: there I seem to get consistently all lipids as expected. Could you share a small snippet of trj where the issue occurs? 

For `-nonflat` only one atom per lipid should be selected

**Comment by ErikPoppleton** [comment]
It appears to happen when you have a raw trajectory and pick whole lipid tails instead of just terminal carbons.  I have attached an example with a 5-frame trajectory where it happens on the 3rd frame.  

[DOPC_example.zip](https://github.com/user-attachments/files/17097313/DOPC_example.zip)

For "one atom per lipid", does that means one atom per tail, or only one atom total?  I switched doing one atom per tail because that's much easier to select (and seems to work?), but I can figure out how to do just one between the two tails as well.

Thanks for the very useful tool!

**Comment by vgapsys** [comment]
Ok, this is indeed due to the fact that `-nonflat` needs a single atom per lipid, i.e. the identity needs to be strictly kept `1 lipid - 1 atom` (not one atom per chain).

In the current example, the initial membrane is setup to be rather flat, so unless it undulates a lot in the simulation, I would not use `-nonflat` flag.

**Comment by ErikPoppleton** [comment]
I notice that when I try to decrease the number of grid points (from 100 to 30 in each direction), the number of lipids included in the `out_over_time.dat` file varies quite substantially.  I am confused about this result, because 30 should be about 1 grid point per lipid, but it seems like somehow it misses lipids if I don't have substantially more grid points than lipids.

**Comment by vgapsys** [comment]
Yes, that might happen if you have too few grid points. The lipids compete for the grid cells, so if a molecule ends up somewhere on the edge of two cells, other lipids may claim all the nearest neighbours and the lipid is not assigned any cell, hence, no property is assigned to it. This issue should not be present with a larger number of grid cells.
